Implementing Shor's algorithm using the Qrisp framework on Cat Qubits.ipynb

```diff
@@ -0,0 +1,770 @@
+    "\n",
+    "This tutorial will walk you through the implementation of the famous Shor's algorithm for prime factoring using Qrisp on the Alice&Bob's Qiskit Provider.\n",
+    "\n",
+    "### Why Shor's prime factoring algorithm?\n",
+    "\n",
+    "A fair question is to ask why it is of any interest to implement Shor's algorithm and run it with a noise model. It all boils down to the quantum advantage that is provided by the quantum algorithm: the current state-of-the-art classical algorithm for prime factoring is the [general number field sieve](https://en.wikipedia.org/wiki/General_number_field_sieve) (GNFS) which, under some assumptions, achieves the prime factoring of a integer in super-polynomial but sub-exponential time. On the contrary, Shor's algorithm promises a polynomial time to achieve the same task. Equivalently said, Shor's algorithm assures a superpolynomial speedup when compared with a classical algorithm.\n",
+    "Published in 1994 by Peter Shor, Shor's algorithm shook the field of theoretical computer science, giving a glance of how powerful quantum computation could be. The implications of successfully running the Shor's algorithm on an actual machine are several, but let's just cite the most hype-generating one: the security of the popular RSA cryptosystem is based on the assumption that prime factorization is computationally hard. Well, that's no longer true with Shor's algorithm.\n",
+    "\n",
+    "### Why Cat Qubits?\n",
+    "\n",
+    "However, we are far from running the algorithm efficiently on a quantum computer, since qubits and operations between them are noisy; at the moment, the information stored in the real hardware is completaly lost after few operations. People in the field are frenatically trying to solve this problem by designing new and more efficient [quantum error correction codes](https://en.wikipedia.org/wiki/Quantum_error_correction) that are supposed to detect and correct errors. In the quantum realm, the errors on a elementary piece of information are bidimensional: you can have bit-flips ( $|0\\rangle \\leftrightarrow |1\\rangle$ ) and phase-flips ( $|+\\rangle \\leftrightarrow |-\\rangle$ ). Luckily, there exists a technology that is able to take care of one source of error, by exponentially subpressing the bit-flip rate, virtually allowing to exclusively deal with the other type of error: the cat qubits architecture. We have a whole [series of tutorial](https://github.com/Alice-Bob-SW/emulation-examples/blob/main/0%20-%20An%20introduction%20to%20cat%20qubits.ipynb) if you want to know more about it.\n",
+    "\n",
+    "Shor's algorithm requires considerably low error rates to give proper results - we will talk about numbers in a following section, when we will do a sketch of the resource estimation of the algorithm - and therefore error correction is needed. So we tought - wait a second, why don't we use cat qubits then, which are the qubits of the future?\n",
```

Comment:
Could you be maybe less casual with the phrasing to be aligned with other notebooks? Instead of " so we though...", something like "It is then a great opportunity to make advantage of cat qubits"

Also, I don't really like the "we will talk about in the next section" part, why not just wait for people to read the section just below? 😅

Reply:
I'm not really sure about the tone, I guess a more serious phrasing could align better with the other notebooks but it's less appealing and this is a larger notebook than the others (btw this " so we though..." is a reference to this https://www.youtube.com/watch?v=zhl-Cs1-sG4).

Mentioning that you will talk about something is conventional if the text has quite an extension, but I can rephrase that